app/Http/Controllers/Auth/AuthController.php

```diff
@@ -11,115 +13,155 @@
-	 *
-	 * @param array $data        	
-	 * @return User
-	 */
-	protected function create(array $data) {
-		return User::create([
-                'email' => $data['email'],
-            		  'username' => $data['name'],
-                'password' => bcrypt($data['password']),
-            		'status' =>TRUE,
-            		'profile_state' => FALSE
-            ]);
-        }
+  /*
+   * |--------------------------------------------------------------------------
+   * | Registration & Login Controller
+   * |--------------------------------------------------------------------------
+   * |
+   * | This controller handles the registration of new users, as well as the
+   * | authentication of existing users. By default, this controller uses
+   * | a simple trait to add these behaviors. Why don't you explore it?
+   * |
+   */
+  
+  use AuthenticatesAndRegistersUsers, ThrottlesLogins;
+  protected $loginPath = '/login';
+  protected $registerPath = '/register';
+  protected $redirectPath = '/';
+  /**
+   * Create a new authentication controller instance.
+   *
+   * @return void
+   */
+  public function __construct() {
+    $this->middleware ( 'guest', [ 
+        'except' => 'getLogout' 
+    ] );
+  }
+  public function socialLogin(SocialAuthenticateUser $authenticateUser, Request $request, $provider = null) {
+    $social_providders = array (
+        "facebook",
+        "google" 
+    );
+    if (in_array ( strtolower ( $provider ), $social_providders )) {
+      return $authenticateUser->execute ( $request, $this, $provider );
+    } else {
+      return redirect ( $this->registerPath )->withErrors ( 'Invalid Login Provider' );
     }
+  }
+  /**
+   * Authenticate users.
+   *
+   * @param array $request          
+   * @return User
+   */
+  public function doLogin(Request $request) {
+    
+    // If the class is using the ThrottlesLogins trait, we can automatically throttle
+    // the login attempts for this application. We'll key this by the username and
+    // the IP address of the client making these requests into this application.
+    $throttles = $this->isUsingThrottlesLoginsTrait ();
+    
+    if ($throttles && $this->hasTooManyLoginAttempts ( $request )) {
+      return $this->sendLockoutResponse ( $request );
+    }
+    
+    $credentials = $this->getCredentials ( $request );
+    $field = (filter_var ( $credentials ['email'], FILTER_VALIDATE_EMAIL )) ? "email" : "username";
+    if (Auth::attempt ( [ 
+        $field => $credentials ['email'],
+        'password' => $credentials ['password'],
+        'status' => TRUE 
+    ], $request->has ( 'remember' ) )) {
+      return $this->handleUserWasAuthenticated ( $request, $throttles );
+    }
+    
+    // If the login attempt was unsuccessful we will increment the number of attempts
+    // to login and redirect the user back to the login form. Of course, when this
+    // user surpasses their maximum number of attempts they will get locked out.
```

Comment:
Let this be a multi-line comment, to make it consistent with the other comment styles